plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -1153,11 +1154,25 @@ public void createMaterializedView(
                     encodeMaterializedViewData(fromConnectorMaterializedViewDefinition(definition)),
                     isUsingSystemSecurity ? null : session.getUser(),
                     createMaterializedViewProperties(session, storageMetadataLocation));
-            if (existing.isPresent()) {
-                updateTable(viewName.getSchemaName(), materializedViewTableInput);
+            try {
+                if (existing.isPresent()) {
+                    updateTable(viewName.getSchemaName(), materializedViewTableInput);
+                }
+                else {
+                    createTable(viewName.getSchemaName(), materializedViewTableInput);
+                }
             }
-            else {
-                createTable(viewName.getSchemaName(), materializedViewTableInput);
+            catch (RuntimeException e) {
+                try {
+                    dropMaterializedViewStorage(fileSystemFactory.create(session), storageMetadataLocation.toString());
```

Comment:
sorry if this is a silly question but why dropping storage table when update failed? I understand doing it when create failed, but for update im not so sure.

Reply:
So the case here is when you do a `CREATE OR REPLACE MV` and there is a MV with that name already. In that case we create a new storage table. If the update fails, the new storage table is cleaned up here because it is not referenced by the MV.

There is something missing here though, when you do a replace the old storage table is not cleaned up on a successful operation. Should do that here.